src/erlcloud_cognito.erl

```diff
@@ -0,0 +1,915 @@
+
+-spec update_identity_provider(binary(), binary()) ->
+    {ok, map()} | {error, any()}.
+update_identity_provider(UserPoolId, ProviderName) ->
+    update_identity_provider(UserPoolId, ProviderName, #{}).
+
+-spec update_identity_provider(binary(), binary(), map()) ->
+    {ok, map()} | {error, any()}.
+update_identity_provider(UserPoolId, ProviderName, ProviderDetails) ->
+    update_identity_provider(UserPoolId, ProviderName, ProviderDetails, #{}).
+
+-spec update_identity_provider(binary(), binary(), map(), map()) ->
+    {ok, map()} | {error, any()}.
+update_identity_provider(UserPoolId, ProviderName,
+                         ProviderDetails, AttributeMapping) ->
+    update_identity_provider(UserPoolId, ProviderName,
+                             ProviderDetails, AttributeMapping, []).
+
+-spec update_identity_provider(binary(), binary(), map(), map(), list()) ->
+    {ok, map()} | {error, any()}.
+update_identity_provider(UserPoolId, ProviderName,
+                         ProviderDetails, AttributeMapping, IdpIdentifiers) ->
+    Config = erlcloud_aws:default_config(),
+    update_identity_provider(UserPoolId, ProviderName, ProviderDetails,
+                             AttributeMapping, IdpIdentifiers, Config).
+
+-spec update_identity_provider(binary(), binary(), map(), map(),
+                               list(), aws_config()) ->
+    {ok, map()} | {error, any()}.
+update_identity_provider(UserPoolId, ProviderName, ProviderDetails,
+                         AttributeMapping, IdpIdentifiers, Config) ->
+    Mandatory = #{
+        <<"UserPoolId">>       => UserPoolId,
+        <<"ProviderName">>     => ProviderName
+    },
+    Optional = #{
+        <<"ProviderDetails">>  => ProviderDetails,
+        <<"AttributeMapping">> => AttributeMapping,
+        <<"IdpIdentifiers">>   => IdpIdentifiers
+    },
+    request(Config, "UpdateIdentityProvider", make_request_body(Mandatory, Optional)).
+
+%%------------------------------------------------------------------------------
+%% Internal Functions
+%%------------------------------------------------------------------------------
+request(Config, Request) ->
+    Result = erlcloud_retry:request(Config, Request, fun handle_result/1),
+    case erlcloud_aws:request_to_return(Result) of
+        {ok, {_, <<>>}}     -> {ok, #{}};
+        {ok, {_, RespBody}} -> {ok, jsx:decode(RespBody, [return_maps])};
+        {error, _} = Error  -> Error
+    end.
+
+request(Config0, OperationName, Request) ->
+    case erlcloud_aws:update_config(Config0) of
+        {ok, Config} ->
+            Body       = jsx:encode(Request),
+            Operation  = "AWSCognitoIdentityProviderService." ++ OperationName,
+            Headers    = get_headers(Config, Operation, Body),
+            AwsRequest = #aws_request{service         = 'cognito-idp',
+                                      uri             = get_url(Config),
+                                      method          = post,
```

Comment:
I wonder if all API requests are actually POSTs, even like `AdminGetUser`?

Reply:
it looks okay, at least it works with any other services like that 🤷

Reply:
As far as i know it is a post